sky/spot/controller.py

```diff
@@ -74,6 +75,36 @@ def __init__(self, job_id: int, dag_yaml: str,
                 job_id_env_vars)
             task.update_envs(task_envs)
 
+    def _download_log_and_stream(self, handle):
+        """ Download the logs from spot cluster, and stream them
+        
+        We do not stream the logs from the spot cluster directly, as the donwload
+        and stream should be faster, as more reliable to preemptions or ssh
+        disconnection during the streaming.
+        """
+        spot_job_logs_dir = os.path.join(
+            constants.SKY_LOGS_DIRECTORY, 'spot_jobs')
+        os.makedirs(spot_job_logs_dir, exist_ok=True)
+        try:
+            log_dirs = self._backend.sync_down_logs(
+                handle,
+                # Download the log for the latest job.
+                job_ids=None,
+                local_dir=spot_job_logs_dir)
+        except exceptions.CommandError as e:
+            logger.info(f'Failed to download the logs: '
+                        f'{common_utils.format_exception(e)}')
+        else:
+            log_dir = list(log_dirs.values())[0]
+
+            # Print the logs to the console.
+            for log_file in os.listdir(log_dir):
```

Comment:
Q: what is the layout of ~/sky_logs/spot_jobs/? Does it have a subdir or a file per spot job id (that has failed)? Wondering why we're printing every *.log file here.